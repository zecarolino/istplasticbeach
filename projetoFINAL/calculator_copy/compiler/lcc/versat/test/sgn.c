#define one() asm("\tldi 0x31\n\twrw CPRT_BASE\n")
#define zero() asm("\tldi 0x30\n\twrw CPRT_BASE\n")
#define nl() asm("\tldi 0xA\n\twrw CPRT_BASE\n")

void compare(int a, int b) {
	if (a == b) one(); else zero();
	if (a != b) one(); else zero();
	if (a <  b) one(); else zero();
	if (a <= b) one(); else zero();
	if (a >  b) one(); else zero();
	if (a >= b) one(); else zero();
	nl();
}

int main() {
	compare(-12,0);		/* 011100 */
	compare(10,15);		/* 011100 */
	compare(15,10);		/* 010011 */
	compare(10,10);		/* 100101 */
	compare(-10,15);	/* 011100 */
	compare(10,-15);	/* 010011 */

	compare(2147483647,15);
	compare(10,2147483647);

	return 0;
}
