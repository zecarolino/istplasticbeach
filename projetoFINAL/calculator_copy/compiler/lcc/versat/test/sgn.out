VCD info: dumpfile xtop.vcd opened for output.
011100
011100
010011
100101
011100
010011
010011
011100
0
Catched Trap at time  2225 clock cycles:
